cmd/opm/serve/serve.go

```diff
@@ -59,12 +73,37 @@ will not be reflected in the served content.
 	}
 
 	cmd.Flags().BoolVar(&s.debug, "debug", false, "enable debug logging")
-	cmd.Flags().StringVarP(&s.port, "port", "p", "50051", "port number to serve on")
 	cmd.Flags().StringVarP(&s.terminationLog, "termination-log", "t", "/dev/termination-log", "path to a container termination log file")
+	cmd.Flags().StringVarP(&s.port, "port", "p", "50051", "port number to serve on")
+	cmd.Flags().BoolVar(&s.pprofEnabledHttp, "pprof-enable-endpoint", false, "enable pprof HTTP endpoint")
+	cmd.Flags().BoolVar(&s.pprofEnabledFile, "pprof-enable-file", false, "enable pprof profile file writing to /tmp/serve.[cpu|mem].prof")
 	return cmd
 }
 
 func (s *serve) run(ctx context.Context) error {
+
+	if s.pprofEnabledHttp {
+		// pprof tool HTTP endpoint
+		go func() {
+			s.logger.Infof("starting pprof endpoint at %s:%s", "localhost", defaultProfPort)
+			http.ListenAndServe("localhost:"+defaultProfPort, nil)
+		}()
+	}
+	if s.pprofEnabledFile {
+		// cpu use profiling data
+		f, err := os.Create(defaultProfCpuFilename)
+		if err != nil {
+			s.logger.WithError(err).Fatalf("unable to open cpu-profile output file: %q", defaultProfCpuFilename)
+		}
+		defer f.Close()
+		if err := pprof.StartCPUProfile(f); err != nil {
+			s.logger.Fatalf("could not start CPU profile: %v", err)
+		}
+		s.logger.Infof("writing CPU pprof data to %q", defaultProfCpuFilename)
+
+		defer pprof.StopCPUProfile()
```

Comment:
Would it make sense to make this a "startup" profile that we explicitly close just before we start the grpc server (and thus after all of the startup parsing has happened)?

That way we could write the profile data (or host it via an http handler) while still keeping the main server running.